### PR TITLE
Remove empty weeks function is now checkit method.

### PR DESCRIPTION
**Pull request by andrayantelo**

### CODE REVIEW

File: docs/checkit.js [hunk 1]
```diff
@@ -460,6 +460,22 @@ CheckIt.prototype.generateCheckmarks = function(calObj, $div) {
      })
 };
 
+CheckIt.prototype.removeEmptyWeeks = function(calObj, $div) {
+    // Remove empty weeks from the calendar
+        
+    $div.find('.month').find('.week').each( function(index) {
+        var counter = 0;
+        $(this).find('td').each(function(td) {
+            if ($(this).children().hasClass("nill"))
```
Comment (johntyree) [inline docs/checkit.js]: Not that it matters, but "nill" is usually spelled "nil."

File: docs/checkit.js [hunk 2]
```diff
@@ -460,6 +460,22 @@ CheckIt.prototype.generateCheckmarks = function(calObj, $div) {
      })
 };
 
+CheckIt.prototype.removeEmptyWeeks = function(calObj, $div) {
```
Comment (johntyree) [inline docs/checkit.js]: `$div` could have a more descriptive name. This function can't take just any div. It has to be a `$calendarDiv` that contains `.month` and `.week` etc.

File: docs/checkit.js [hunk 3]
```diff
@@ -460,6 +460,22 @@ CheckIt.prototype.generateCheckmarks = function(calObj, $div) {
...
+        var counter = 0;
+        $(this).find('td').each(function(td) {
+            if ($(this).children().hasClass("nill")) 
+            counter +=1;
+        })
+        if (counter === 7) 
+        {
+            $(this).remove();
+        }
+    })
```
Comment (johntyree) [inline docs/checkit.js]: This is a clever way to figure out if the week should be removed or not, but you can also just ask jquery how many things have the `nil` class.

Try something like `$(this).children('.nil').length`

**Review by johntyree — CHANGES_REQUESTED** [review]

File: docs/checkit.js (hunk 1 above)
Comment (andrayantelo) [reply docs/checkit.js]: Gosh darn it. It matters to me.

File: docs/checkit.js (hunk 2 above)
Comment (andrayantelo) [reply docs/checkit.js]: Changed name to $calendarDiv

File: docs/checkit.js [hunk 4]
```diff
@@ -460,17 +458,11 @@ CheckIt.prototype.generateCheckmarks = function(calObj, $div) {
...
         
-    $div.find('.month').find('.week').each( function(index) {
-        var counter = 0;
-        $(this).find('td').each(function(td) {
-            if ($(this).children().hasClass("nill")) 
-            counter +=1;
-        })
-        if (counter === 7) 
-        {
+    $calendarDiv.find('.month').find('.week').each( function(index) {
```
Comment (johntyree) [inline docs/checkit.js]: These are CSS selectors, right? I don't think you should have to do multiple calls in a row. `find('.month > .week')`

File: docs/checkit.js [hunk 5]
```diff
@@ -460,17 +458,11 @@ CheckIt.prototype.generateCheckmarks = function(calObj, $div) {
...
+        if ($(this).find('td').children('.nil').length === 7) {
```
Comment (johntyree) [inline docs/checkit.js]: `.find("td > .nil")`

**Review by johntyree — CHANGES_REQUESTED** [review]

**Review by johntyree — APPROVED** [review]

**Pull request merged**